src/julia-syntax.scm

```diff
@@ -1704,9 +1704,7 @@
                    (else
                     (cf (cdr old-fargs) (cdr old-args)
                         (cons farg new-fargs) (cons arg new-args) renames varfarg vararg))))))
-          (if (and (= (length args) 1) (integer? (car args)))
-              e ; hack: don't compress e.g. f.(1) so that deprecation for getfield works
```

Comment:
It's aways satisfying to see code disappear when it has an associated comment describing it as a "hack." 😉